src/rest_api/amoc_api_logic_handler.erl

```diff
@@ -0,0 +1,84 @@
+
+handle_request('NodesGet', _Req, _Context) ->
+    Status = amoc_cluster:get_status(),
+    Connected = maps:get(connected, Status, []),
+    FailedToConnect = maps:get(failed_to_connect, Status, []),
+    ConnectionLost = maps:get(connection_lost, Status, []),
+    Up = [{Node, <<"up">>} || Node <- [node() | Connected]],
+    DownNodes = lists:usort(FailedToConnect ++ ConnectionLost),
+    Down = [{Node, <<"down">>} || Node <- DownNodes],
+    ResponseList = Up ++ Down,
+    {200, #{}, [{<<"nodes">>, ResponseList}]};
+handle_request('ScenariosGet', _Req, _Context) ->
+    Scenarios = amoc_scenario:list_scenario_modules(),
+    BinaryScenarios = [atom_to_binary(S, latin1) || S <- Scenarios],
+    {200, #{}, [{<<"scenarios">>, BinaryScenarios}]};
+handle_request('StatusGet', _Req, _Context) ->
+    Apps = application:which_applications(),
+    Status = case lists:keyfind(amoc, 1, Apps) of
+                 {amoc, _Desc, _Vsn} -> <<"up">>;
+                 false -> <<"down">>
+             end,
+    {200, #{}, [{<<"node_status">>, Status}]};
+handle_request('ScenariosIdGet', _Req, #{id := Resource}) ->
+    Scenario = binary_to_atom(Resource, latin1),
+    case amoc_api_scenario_status:test_status(Scenario) of
+        doesnt_exist ->
+            {404, #{}, #{}};
+        Status ->
+            BinStatus = atom_to_binary(Status, latin1),
+            {200, #{}, [{<<"scenario_status">>, BinStatus}]}
+    end;
+handle_request('ScenariosIdPatch', _Req, #{'ScenarioExecution' := Body,
+                                           id := Resource }) ->
+    Scenario = binary_to_atom(Resource, latin1),
+    case amoc_api_scenario_status:test_status(Scenario) of
+        doesnt_exist ->
+            {404, #{}, #{}};
+        loaded ->
+            Users = maps:get(<<"users">>, Body),
+            SettingsMap = maps:get(<<"settings">>, Body, #{}),
+            Settings = [begin
+                            Key = binary_to_atom(K, latin1),
+                            {ok, Value} = amoc_config_env:parse_value(V),
+                            {Key, Value}
+                        end || {K, V} <- maps:to_list(SettingsMap)],
+            case amoc_dist:do(Scenario, Users, Settings) of
+                {ok, _} -> {200, #{}, [{<<"scenario">>, <<"started">>}]};
+                {error, _} -> {200, #{}, [{<<"scenario">>, <<"error">>}]}
+            end;
+        _ ->
+            {200, #{}, [{<<"scenario">>, <<"error">>}]}
+    end;
+handle_request('ScenariosUploadPut', Req, _Context) ->
+    {ok, ModuleSource, _} = cowboy_req:read_body(Req),
+    case amoc_api_upload_scenario:upload(ModuleSource) of
+        ok ->
+            {200, #{}, [{<<"compile">>, <<"ok">>}]};
+        {error, invalid_module} ->
+            {400, #{}, [{<<"error">>, <<"invalid module">>}]};
+        {error, Error} ->
+            {200, #{}, [{<<"compile">>, Error}]}
+    end;
+handle_request(OperationID, Req, Context) ->
+    error_logger:error_msg(
```

Comment:
In amoc we are using the built-in Erlang logger. I think it'd be better to be consistent with that and use it here as well.

Reply:
sure, that is a copy-past from the default handler. thanks for catching that.